src/Symfony/Component/Routing/RequestContext.php

```diff
@@ -43,6 +43,7 @@ public function __construct(string $baseUrl = '', string $method = 'GET', string
         $this->setHttpsPort($httpsPort);
         $this->setPathInfo($path);
         $this->setQueryString($queryString);
+        $this->setParameters([...$this->parameters, ...$parameters]);
```

Comment:
The class is not final, so we merge it with eventually parent set parameters.

Reply:
we could also create a deprecation `if (count($this->parameters)) {` and simplify this in Symfony 8?